src/com/maddyhome/idea/vim/ui/ExEntryPanel.java

```diff
@@ -302,13 +307,50 @@ public void componentResized(ComponentEvent e) {
     protected void textChanged(@NotNull DocumentEvent e) {
       final Editor editor = entry.getEditor();
 
-      final boolean forwards = !label.getText().equals("?");
-      final String searchText = entry.getActualText();
+      LineRange searchRange = null;
+      char separator = label.getText().charAt(0);
+      String searchText = entry.getActualText();
+      if (label.getText().equals(":")) {
+        final ExCommand command = getIncsearchCommand(searchText);
+        if (command == null) {
+          return;
+        }
+        searchText = "";
+        final String argument = command.getArgument();
+        if (argument.length() > 1) {  // E.g. skip '/' in `:%s/`. `%` is range, `s` is command, `/` is argument
+          separator = argument.charAt(0);
+          searchText = argument.substring(1);
+        }
+        final Ranges ranges = command.getRanges();
+        ranges.setDefaultLine(EditorHelper.offsetToCharacterPosition(editor, caretOffset).line);
+        searchRange = command.getLineRange(editor, entry.getContext());
+      }
+
+      final boolean forwards = !label.getText().equals("?");  // :s, :g, :v are treated as forwards
       final CharPointer p = new CharPointer(searchText);
-      final CharPointer end = RegExp.skip_regexp(new CharPointer(searchText), forwards ? '/' : '?', true);
+      final CharPointer end = RegExp.skip_regexp(new CharPointer(searchText), separator, true);
       final String pattern = p.substring(end.pointer() - p.pointer());
 
-      VimPlugin.getSearch().updateIncsearchHighlights(editor, pattern, forwards, caretOffset);
+      VimPlugin.getSearch().updateIncsearchHighlights(editor, pattern, forwards, caretOffset, searchRange);
+    }
+
+    @Nullable
+    private ExCommand getIncsearchCommand(String commandText) {
+      try {
+        final ExCommand exCommand = CommandParser.getInstance().parse(commandText);
+        final String command = exCommand.getCommand();
+        if (command.equals("s") || command.equals("substitute")
```

Comment:
Substitute command (and others) can be called using each amount of characters from short-size to full-size. So, `su`, `sub`, `subs`, etc. are also valid as well.

Reply:
Aha, I didn't know that. I'll get it fixed.